### PR TITLE
Second commit

### PR DESCRIPTION
**Pull request by odebretseni**

### CODE REVIEW

File: meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  company: "TenantCloud LLC"
+
+  license: BSD
+
+  min_ansible_version: 2.4
+
+  platforms:
+    - name: MacOS
+      versions:
+        - "Catalina"
```
Comment (BrunIF) [inline meta/main.yml]: Mac OS version should be a number

File: tasks/change-env-file.yml [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+  lineinfile:
+    path: /Users/{{ work_user }}/{{ work_dir }}/{{ dashboard_dir }}/.env
+    regexp: 'QUEUE_DRIVER='
+    line: 'QUEUE_DRIVER=redis'
+
+- name: Add SOCKETIO_URL
+  lineinfile:
+    path: /Users/{{ work_user }}/{{ work_dir }}/{{ dashboard_dir }}/.env
+    regexp: '^SOCKETIO_URL='
+    line: 'SOCKETIO_URL=https://home.{{ work_domain }}'
```
Comment (BrunIF) [inline tasks/change-env-file.yml]: What happens if you raise a different project?  Keller

File: tasks/change-env-file.yml [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    regexp: '^DB_USERNAME='
+    line: 'DB_USERNAME={{ mysql_user }}'
+
+- name: Add DB_PASSWORD
+  lineinfile:
+    path: /Users/{{ work_user }}/{{ work_dir }}/{{ dashboard_dir }}/.env
+    regexp: '^DB_PASSWORD='
+    line: 'DB_PASSWORD='
+
+- name: Add DB_ACTIVITY_LOG_HOST
```
Comment (BrunIF) [inline tasks/change-env-file.yml]: needless. It is deprecated now.

File: tasks/change-env-file.yml [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    path: /Users/{{ work_user }}/{{ work_dir }}/{{ dashboard_dir }}/.env
+    regexp: '^CATEGORY_CREATE_ENABLED='
+    line: 'CATEGORY_CREATE_ENABLED=true'
+
+- name: Add S3_MOBILE_REGION
+  lineinfile:
+    path: /Users/{{ work_user }}/{{ work_dir }}/{{ dashboard_dir }}/.env
+    regexp: '^S3_MOBILE_REGION='
+    line: 'S3_MOBILE_REGION=us-east-1'
+
```
Comment (vikfedkiv) [inline tasks/change-env-file.yml]: Add CDN_URL=S3_AWS_URL=https://home.{{ work_domain }}:9002/local

File: tasks/create-minio-backet.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  get_url:
+    url: https://dl.min.io/client/mc/release/darwin-amd64/mc
+    dest: /usr/local/bin/minio-client
+    mode: '0777'
+
+- name: Create s3 local project
+  command: /usr/local/bin/minio-client config host add s3 https://home.tenantcloud.dev:9002 minio miniostorage
+  changed_when: false
+
+- name: Create s3 local backet
```
Comment (vikfedkiv) [inline tasks/create-minio-backet.yml]: Bucket "local" is present in docker image on ECR

File: tasks/change-env-file.yml [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    regexp: '^AWS_LAMBDA_REGION='
+    line: 'AWS_LAMBDA_REGION={{ aws_lambda_region }}'
+
+- name: Add AWS_LAMBDA_S3
+  lineinfile:
+    path: /Users/{{ work_user }}/{{ work_dir }}/{{ dashboard_dir }}/.env
+    regexp: '^AWS_LAMBDA_S3='
+    line: 'AWS_LAMBDA_S3={{ aws_lambda_s3 }}'
+
+- name: Add CONVERTER_DRIVER
```
Comment (BrunIF) [inline tasks/change-env-file.yml]: You install LibreOffice, or get it in a container?

File: tasks/copy-idea-settings.yml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Create idea conf directory
+  file:
+    path: /Users/{{ work_user }}/{{ work_dir }}/{{ dashboard_dir }}/.idea
+    state: directory
+    mode: 0777
+  become: false
```
Comment (BrunIF) [inline tasks/copy-idea-settings.yml]: By default `become` is false. 
Don`t run ansible playbook with key `-k -K`

File: tasks/copy-pre-commit-script.yml [hunk 8]
```diff
@@ -0,0 +1,18 @@
+---
+#- name: pre-commit copy
```
Comment (BrunIF) [inline tasks/copy-pre-commit-script.yml]: Delete old code

File: tasks/create-minio-backet.yml [hunk 9]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (BrunIF) [inline tasks/create-minio-backet.yml]: needless. Use our docker image. 
Create bucket with `docker exec .....`

File: tasks/create-work-folder.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    state: directory
+    mode: 0777
+  become: false
+
+- name: Create nginx ssl directory
+  file:
+    path: /usr/local/etc/nginx/ssl
+    state: directory
+    mode: 0777
+  become: false
```
Comment (BrunIF) [inline tasks/create-work-folder.yml]: needless.
Review all files

File: tasks/create-work-folder.yml [hunk 11]
```diff
@@ -0,0 +1,42 @@
+---
```
Comment (BrunIF) [inline tasks/create-work-folder.yml]: It is not logical. You create directory in one place, and work with them in other tasks

File: tasks/create_db_mysql.yml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+---
+- name: Start MySQL
+  command: /usr/local/bin/brew services start mysql@5.7
+  changed_when: false
+
+- name: Export MySQL path
+  command: echo 'export PATH="/usr/local/opt/mysql@5.7/bin:$PATH"' >> ~/.zshrc
```
Comment (BrunIF) [inline tasks/create_db_mysql.yml]: MySQL version to `default/main.yml`
What happend if we will use MySQL 8?

File: tasks/docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,49 @@
+---
```
Comment (BrunIF) [inline tasks/docker-compose.yml:1]: This is an old configuration. Please review. 
I don't understand what do you do.

File: tasks/git-clone.yml [hunk 14]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (BrunIF) [inline tasks/git-clone.yml]: It is not logical.
Please do next.
Clone -> Configure `.env` -> Setup project

File: tasks/main.yml [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+- include: change-env-file.yml
+- include: change-nginx-settings.yml
+- include: fix-php-settings.yml
+- include: docker-compose.yml
+- include: create-minio-backet.yml
+- include: create_db_mysql.yml
+- include: copy-pre-commit-script.yml
+- include: install-and-run-composer.yml
+- include: run_yarn.yml
+- include: run_dnsmasq.yml
```
Comment (BrunIF) [inline tasks/main.yml]: Configure?

File: tasks/main.yml [hunk 16]
```diff
@@ -0,0 +1,14 @@
+---
+- include: create-work-folder.yml
+- include: git-clone.yml
+- include: copy-idea-settings.yml
+- include: change-env-file.yml
```
Comment (BrunIF) [inline tasks/main.yml]: Configure?

File: tasks/main.yml [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+- include: change-nginx-settings.yml
```
Comment (BrunIF) [inline tasks/main.yml]: Configure?

File: templates/ext-xdebug.ini [hunk 18]
```diff
@@ -0,0 +1,5 @@
+[xdebug]
+#zend_extension=<full_path_to_xdebug_extension>
+xdebug.remote_enable=1
+xdebug.remote_host=localhost
```
Comment (BrunIF) [inline templates/ext-xdebug.ini:4]: Better will be use IP address

File: templates/home.kellermanagement.dev.conf [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+    # you can tell the browser that it can only download content from the domains you explicitly allow
+    # http://www.html5rocks.com/en/tutorials/security/content-security-policy/
+    # https://www.owasp.org/index.php/Content_Security_Policy
+    # I need to change our application code so we can increase security by disabling 'unsafe-inline' 'unsafe-eval'
+    # directives for css and js(if you have inline css or js, you will need to keep it too).
+    # more: http://www.html5rocks.com/en/tutorials/security/content-security-policy/#inline-code-considered-harmful
+    #add_header Content-Security-Policy "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://ssl.google-analytics.com https://assets.zendesk.com https://connect.facebook.net; img-src 'self' https://ssl.google-analytics.com https://s-static.ak.facebook.com https://assets.zendesk.com; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://assets.zendesk.com; font-src 'self' https://themes.googleusercontent.com; frame-src https://assets.zendesk.com https://www.facebook.com https://s-static.ak.facebook.com https://tautt.zendesk.com; object-src 'none'";
+
+    location /socket.io/ {
```
Comment (BrunIF) [inline templates/home.kellermanagement.dev.conf]: deprecated

File: templates/home.kellermanagement.dev.conf [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+
+    location ~ \.php$ {
+        fastcgi_split_path_info ^(.+\.php)(/.+)$;
+        fastcgi_pass 127.0.0.1:9000;
+        fastcgi_index index.php;
+        include fastcgi_params;
+        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+    }
+
+    access_log /usr/local/var/log/nginx/home.km.access.log;
```
Comment (BrunIF) [inline templates/home.kellermanagement.dev.conf]: `home.km`?

File: templates/home.tenantcloud.dev.conf [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+    # you can tell the browser that it can only download content from the domains you explicitly allow
+    # http://www.html5rocks.com/en/tutorials/security/content-security-policy/
+    # https://www.owasp.org/index.php/Content_Security_Policy
+    # I need to change our application code so we can increase security by disabling 'unsafe-inline' 'unsafe-eval'
+    # directives for css and js(if you have inline css or js, you will need to keep it too).
+    # more: http://www.html5rocks.com/en/tutorials/security/content-security-policy/#inline-code-considered-harmful
+    #add_header Content-Security-Policy "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://ssl.google-analytics.com https://assets.zendesk.com https://connect.facebook.net; img-src 'self' https://ssl.google-analytics.com https://s-static.ak.facebook.com https://assets.zendesk.com; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://assets.zendesk.com; font-src 'self' https://themes.googleusercontent.com; frame-src https://assets.zendesk.com https://www.facebook.com https://s-static.ak.facebook.com https://tautt.zendesk.com; object-src 'none'";
+
+    location /socket.io/ {
```
Comment (BrunIF) [inline templates/home.tenantcloud.dev.conf:73]: deprecated

File: templates/home.tenantcloud.dev.conf [hunk 22]
```diff
@@ -0,0 +1,103 @@
+server {
```
Comment (BrunIF) [inline templates/home.tenantcloud.dev.conf:1]: templates/home.conf

File: templates/mysql.sh [hunk 23]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (BrunIF) [inline templates/mysql.sh:1]: Better practice https://docs.ansible.com/ansible/latest/modules/mysql_db_module.html#mysql-db-module

File: templates/yarn.sh [hunk 24]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (BrunIF) [inline templates/yarn.sh]: https://docs.ansible.com/ansible/latest/modules/npm_module.html

File: templates/yarn.sh [hunk 25]
```diff
@@ -0,0 +1,15 @@
...
+
+source ~/.zshrc
+
+cd /Users/{{ work_user }}/{{ work_dir }}/{{ dashboard_dir }}
+
+/Users/{{ work_user }}/.yarn/bin/yarn install --frozen-lockfile
+
+/Users/{{ work_user }}/.yarn/bin/yarn release
+
+cp tests/Protractor/protractorEnvironment.example.js tests/Protractor/protractorEnvironment.js
```
Comment (BrunIF) [inline templates/yarn.sh]: This is another task

File: vars/main.yml [hunk 26]
```diff
@@ -0,0 +1,30 @@
+---
+# vars file for tenantcloud.ansible_role_dashboard
+
+work_user: "user"
+ansible_python_interpreter: "/usr/local/bin/python3"
+dashboard_git:
```
Comment (BrunIF) [inline vars/main.yml]: `dashboard_git: ""`

**Review by BrunIF — CHANGES_REQUESTED** [review]

File: tasks/change-env-file.yml (hunk 6 above)
Comment (odebretseni) [reply tasks/change-env-file.yml]: install separately

File: templates/home.tenantcloud.dev.conf (hunk 22 above)
Comment (odebretseni) [reply templates/home.tenantcloud.dev.conf:1]: and keller?

**Review by BrunIF — APPROVED** [review]

**Pull request merged**